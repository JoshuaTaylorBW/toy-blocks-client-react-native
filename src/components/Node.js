import PropTypes from "prop-types";
import { View, Text, TouchableOpacity } from "react-native";
import styled from "styled-components";
import colors from "../constants/colors";
import Status from "./Status";

const Wrapper = styled.TouchableOpacity`
  margin: 10px;
  padding: 10px;
  border: 1px solid ${colors.border};
  background: ${colors.contentBackground};
  border-radius: 5px;
`;

const Head = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

const Body = styled.View`
  display: flex;
  justify-content: space-between;
`;

const Name = styled.Text`
  margin-bottom: 6px;
  color: ${colors.primary};
  font-size: 16px;
  font-weight: 700;
`;

const URL = styled.Text`
  margin-top: 2px;
  margin-bottom: 2px;
  color: ${colors.faded};
  font-weight: 500;
`;

const Toolbar = styled.View`
  display: flex;
  justify-content: flex-end;
`;

const BlocksWrapper = styled.View`
  margin: 20px 0;
  text-align: center;
  color: ${colors.danger};
  font-weight: 700;
`;

const Node = ({ node, expanded, toggleNodeExpanded }) => (
  <Wrapper onPress={() => toggleNodeExpanded(node)}>
    <Head>
      <Name>{node.name || "Unknown"}</Name>
      <Toolbar>
        <Status {...node} />
      </Toolbar>
    </Head>
    <Body>
      <URL>{node.url}</URL>
    </Body>
    {expanded && (
      <BlocksWrapper>
        <Text>Blocks go here</Text>
      </BlocksWrapper>
    )}
  </Wrapper>
);

Node.propTypes = {
  node: PropTypes.shape({
    url: PropTypes.string,
    online: PropTypes.bool,
    name: PropTypes.string,
    loading: PropTypes.bool
  }).isRequired,
  expanded: PropTypes.bool,
  toggleNodeExpanded: PropTypes.func.isRequired
};

export default Node;
