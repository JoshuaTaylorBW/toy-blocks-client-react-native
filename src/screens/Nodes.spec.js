import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedNodes, { Nodes } from "./Nodes";
import Node from "../components/Node";

describe("<Nodes />", () => {
  const actions = {
    checkNodeStatuses: jest.fn()
  };

  const nodes = {
    list: [
      {
        url: "https://thawing-springs-53971.herokuapp.com",
        online: false,
        name: "Node 1",
        loading: false
      },
      {
        url: "https://secret-lowlands-62331.herokuapp.com",
        online: false,
        name: "Node 2",
        loading: false
      }
    ]
  };

  it("should contain <Node />", () => {
    const wrapper = shallow(<Nodes actions={actions} nodes={nodes} />);

    expect(wrapper.find(Node).length).toEqual(2);
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({ nodes });
    const component = create(
      <Provider store={store}>
        <ConnectedNodes />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
