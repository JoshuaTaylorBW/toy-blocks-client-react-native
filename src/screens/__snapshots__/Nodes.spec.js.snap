// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nodes /> should match snapshot 1`] = `
<div>
  <h1>
    Nodes
  </h1>
  <div
    className="sc-htpNat cAglkR"
  >
    <div
      className="sc-bxivhb gSSgMV"
    >
      <div
        className="sc-EHOje fvoECt"
      >
        Node 1
      </div>
      <div
        className="sc-gzVnrw hwzuqy"
      >
        <div
          className="sc-bdVaJa ffSlWY"
          color="danger"
        >
          Offline
        </div>
        <button
          className="sc-htoDjs essTOS"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="sc-ifAKCX fOGIFb"
    >
      <div
        className="sc-bZQynM dSZNti"
      >
        https://thawing-springs-53971.herokuapp.com
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat cAglkR"
  >
    <div
      className="sc-bxivhb gSSgMV"
    >
      <div
        className="sc-EHOje fvoECt"
      >
        Node 2
      </div>
      <div
        className="sc-gzVnrw hwzuqy"
      >
        <div
          className="sc-bdVaJa ffSlWY"
          color="danger"
        >
          Offline
        </div>
        <button
          className="sc-htoDjs essTOS"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="sc-ifAKCX fOGIFb"
    >
      <div
        className="sc-bZQynM dSZNti"
      >
        https://secret-lowlands-62331.herokuapp.com
      </div>
    </div>
  </div>
</div>
`;
